user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -231,6 +231,25 @@ define([
     $folder.find("> .element .name").trigger("click");
   };
 
+  FileBrowser.clearTreeCache = function () {
+    let url = CONTEXT_PATH + "plugin/scheduler-plugin/api/generic-files/tree/cache";
+
+    $.ajax({
+      async: true,
```

Comment:
By being an async request, and not waiting for the response, it's not guaranteed that the follow-up requests reach the server and are processed by it first. This could make a follow-up tree retrieval to return a cached result.
You could change to `async: false`, given that we are considering and will likely be removing server cache anyway, thus the sync request representing only a temporary hack.